### PR TITLE
#14: Send data to ContactHub (closes #14)

### LINKED ISSUE #14: Send data to ContactHub
[Project card](https://github.com/buildo/socialone/projects/1?fullscreen=true#card-4336822)

## requirements
send the retrieved user data to the owner's ContactHub account

### PR DESCRIPTION
**Pull request by FrancescoCioria**
Closes #14

### tests performed
- `npm start`
- open console
- click on any of the login button
- user data sent successfully to contacthub 🎉 

> An example of partial test is a field displaying 4 possible values. If 3 values are tested, with screenshots, and 1 is not, then it should be mentioned here.}

### CODE REVIEW

**Comment by FrancescoCioria** [comment]
@dhinus note: I'm improving tests and validation in general in #16

File: demo/index.ejs [hunk 1]
```diff
@@ -44,6 +53,7 @@
           google: '795655891034-5tou4pabjpdjdcmunv1970ujkuqggj7m.apps.googleusercontent.com',
           linkedin: '778rorytckysk3'
         },
+        contacthub: window.ch,
```
Comment (FrancescoCioria) [inline demo/index.ejs]: @dhinus this is the interesting part: `SocialAutofill` requires a direct pointer to the sdk
(I find it simpler to use both for us and the user compared to an API where we ask for the global name or something like that)

File: src/socialNetworks/index.ts [hunk 2]
```diff
@@ -15,28 +15,26 @@ export default class SocialNetworks {
...
-
-        const user = getUserFromFacebookUser(facebookUser);
-
-        console.log(user);
-      }, e => {
-        alert('Whoops! ' + e.error.message);
-      });
-    }, e => {
-      console.error(e)
+    return new Promise((resolve, reject) => {
```
Comment (FrancescoCioria) [inline src/socialNetworks/index.ts]: I already opened an issue to add a polyfill for `Promise` ( #17 )

File: src/models/contacthub-sdk-browser.ts [hunk 3]
```diff
@@ -44,7 +44,7 @@ export type CustomerTags = {
 };
 
 export type CustomerContacts = {
-  email?: string,
+  email: string,
```
Comment (dhinus) [inline src/models/contacthub-sdk-browser.ts]: this is not always required, it depends on workspace-specific settings in Hub.

File: src/models/contacthub-sdk-browser.ts [hunk 4]
```diff
@@ -91,7 +91,7 @@ export type CustomerBase = {
   dob?: string,
   locale?: string,
   timezone?: string,
-  contacts?: CustomerContacts,
+  contacts: CustomerContacts,
```
Comment (dhinus) [inline src/models/contacthub-sdk-browser.ts]: same as above

File: src/userConverters/index.ts [hunk 5]
```diff
@@ -7,13 +7,16 @@ import {
 
 export function getUserFromFacebookUser(facebookUser: FacebookUser): User {
   return {
-    id: facebookUser.id,
+    externalId: facebookUser.id,
```
Comment (dhinus) [inline src/userConverters/index.ts]: `externalId` should not be used to store social network IDs, tracked in #21

**Pull request merged**